_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -1208,6 +1208,13 @@ mfxStatus ImplementationAvc::Init(mfxVideoParam * par)
         amtMctf = std::make_shared<CMC>();
         MFX_CHECK_NULL_PTR1(amtMctf);
 
+        request.Info        = m_video.mfx.FrameInfo;
+        request.Type        = MFX_MEMTYPE_D3D_INT;
+        request.NumFrameMin = (mfxU16)m_emulatorForSyncPart.GetStageGreediness(AsyncRoutineEmulator::STG_WAIT_MCTF) + 1;
```

Comment:
How many frames do we actually get here?

Reply:
We get 3 frames, found a corner case were we need 3 as the other two are waiting for the encoder to be processed.

Reply:
Wondering, why 3? If we always make mctf->wait() after two mctf->submit() calls , then  why do we need 3rd frame?

Reply:
>> as the other two are waiting for the encoder to be processed.

This is bothering. Can you, please, describe this corner use case? which async depth? which gop structure? did we somehow alter it?